src/lt/objs/deploy.cljs

```diff
@@ -21,31 +21,64 @@
 
 (def shell (load/node-module "shelljs"))
 (def fs (js/require "fs"))
+; Update the comment for the following to describe the full situation:
+(def original-fs (js/require "original-fs")) ;; Required per [fs - reading asar files fails · Issue #1658 · atom/electron](https://github.com/atom/electron/issues/1658)
 (def zlib (js/require "zlib"))
 (def request (load/node-module "request"))
-(def tar (load/node-module "tar"))
+(def tar (load/node-module "tar-fs"))
+(def zip (load/node-module "node-stream-zip"))
 (def home-path (files/lt-home ""))
 ;; TODO: get-proxy
 ;; (def get-proxy (.-App.getProxyForURL (js/require "nw.gui")))
 (def get-proxy)
 
-(defn tar-path [v]
+; TODO: Cleanup platform name code
+
+; TODO: Rename to release-file-extension
+(def tar-extension (if (platform/win?) ".zip" ".tar.gz"))
+
+(defn get-release-download-url
+  "Retrieve the download URL for the specified release version and this computer's platform."
+  [v cb]
+  (let [release-info-url (str "https://api.github.com/repos/LightTable/LightTable/releases/tags/"
+                              v)]
+    (fetch/xhr release-info-url
+               {}
+               (fn [data]
+                 (let [parsed-data (js/JSON.parse data)
+                       assets (.-assets parsed-data)
+                       platform-name (name platform/platform)
+                       platform-asset-name (str "lighttable-" v "-" platform-name tar-extension)
+                       platform-asset (first (filter #(= (.-name %) platform-asset-name)
+                                                     assets))
+                       download-url (.-browser_download_url platform-asset)
+                       ]
+                   (cb download-url)
+                   )))))
+
+;(get-release-download-url "0.8.1" (fn [u] (console/log u)))
+
+(defn tar-path [v cb]
   (if (cache/fetch :edge)
-    (str "https://api.github.com/repos/LightTable/LightTable/tarball/master")
-    (str "https://api.github.com/repos/LightTable/LightTable/tarball/" v)))
+    (cb (str "https://api.github.com/repos/LightTable/LightTable/tarball/master")) ; TODO: Fix to use CI build artifact.
+    (get-release-download-url v cb)))
+
+;(tar-path "0.8.1" (fn [url] (console/log url)))
+
+;
 
 (def version-regex #"^\d+\.\d+\.\d+(-.*)?$")
 
 (defn get-versions []
   (let [vstr (:content (files/open-sync (files/lt-home "core/version.json")))]
     (js->clj (.parse js/JSON vstr) :keywordize-keys true)))
 
-(defn proxy? []
-  (let [p (get-proxy (tar-path "0.5.0"))]
-    (when (str-contains? p "PROXY")
-      (-> p
-          (string/split " ")
-          (second)))))
+;; (defn proxy? []
+;;   (let [p (get-proxy (tar-path "0.5.0"))]
+;;     (when (str-contains? p "PROXY")
+;;       (-> p
+;;           (string/split " ")
+;;           (second)))))
```

Comment:
Note also that I'm commenting out this function entirely. [My original comment about these changes](https://github.com/LightTable/LightTable/pull/2176#discussion_r60169719):

> I commented this out because it calls `tar-path`, but that now requires a callback, and I couldn't find any callers of this function anyways.